libs/sdk-common/src/input_parser.rs

```diff
@@ -183,8 +191,21 @@ use crate::prelude::*;
 pub async fn parse(
     input: &str,
     external_input_parsers: Option<&[ExternalInputParser]>,
+    dns_resolver: Option<&AsyncResolver<GenericConnector<TokioRuntimeProvider>>>,
 ) -> Result<InputType> {
-    let input = input.trim();
+    let mut input = input.trim();
```

Comment:
NIT: it's generally better to avoid mut when possible. Logic is easier to follow when we know values cannot change. In this specific case, I usually prefer to use shadowing (creating a new variable with the same name).